File: solutions/fsharp/armstrong-numbers/1/ArmstrongNumbers.fs
module ArmstrongNumbers

let isArmstrongNumber number =
    match number, (string number).Length with
    | 0, _ -> true
    | _, 1 -> true
    | n, l ->
        n = (n
             |> string
             |> Seq.fold (fun s c -> s + pown (int c - int '0') l) 0)

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number) {
      var chars = number.ToString().ToCharArray();
      var exponent = chars.Length;
      var sum = chars.Aggregate(0, (t, n) => t + (int)Math.Pow(n - '0', exponent));
      
      return sum == number;
    }
}